airflow-core/src/airflow/api_fastapi/core_api/routes/public/plugins.py

```diff
@@ -44,8 +47,22 @@ def get_plugins(
     offset: QueryOffset,
 ) -> PluginCollectionResponse:
     plugins_info = sorted(plugins_manager.get_plugin_info(), key=lambda x: x["name"])
+    valid_plugins: list[PluginResponse] = []
+    for plugin_dict in plugins_info[offset.value :][: limit.value]:
+        try:
+            # Validate each plugin individually
+            plugin = PluginResponse.model_validate(plugin_dict)
+            valid_plugins.append(plugin)
+        except ValidationError as e:
+            logger.warning(
+                "Skipping invalid plugin due to error",
+                plugin_name=plugin_dict.get("name", "<unknown>"),
+                error=str(e),
+            )
+            continue
```

Comment:
Just realized you can't do that. You need to apply the validation before the 'virtual' pagination. 

Otherwise the returned page could be 'partial' which isn't correct.

Filter all `plugins_info` and then apply pagination and `total_entries`.
|